main/src/library/Fixpoint/Compiler.flix

```diff
@@ -268,7 +268,7 @@ namespace Fixpoint {
         Array.foldLeft(acc -> match (atom, rowVar) -> match atom {
             case BodyAtom(_, denotation, Polarity.Positive, _, terms) =>
                 Array.mapWithIndex(term -> i -> (term, i), terms) |>
-                Array.foldRight(match (term, i) -> match term {
+                Array.foldRightOld(match (term, i) -> match term {
```

Comment:
I can't figure out how to make this work with the new `Array.foldRight`.

Reply:
I'm struggling with this last one. It is not
```
Array.foldRight(match (term, i) -> k -> match term {
        ...
            Map.insertWith(_ -> t -> t, var, RamTerm.RowLoad(rowVar, i), k())
        ...
                Map.insertWith(_ -> t -> t, var, RamTerm.RowLoad(rowVar, i), k())
        ...
                Map.insertWith(f, var, RamTerm.LoadLatVar(rowVar), k())
    }
    case _ => x -> x
}, acc)
```
but I'm not sure, what is supposed to be

Reply:
Can you point me to the file / line where this is?

Reply:
It is this line (line 271) that needs to be refactored to use the new `foldRight` instead of the old `foldRightOld`. But I cannot seem to figure out how to make it work. The code in the other comment is from line 271 to line 283

Reply:
The use of the two `match` expressions definitely make this tricky :/

Reply:
Code is crazy :S

Reply:
Proposal: What if you change it to a foldLeft? (You may have to swap the argument order on line 271)

Reply:
Yes, good idea

Reply:
Let me know if this works. (The reason I think we can switch to foldLeft is because the order does not seem to matter, we just building a new map).

Second question, is this the last blocker?

Reply:
I can't simply change it to `Array.foldLeft`. I get the horrible error message, that I cannot resolve
```
>> Unable to unify the types: '(?, ?)' and 'Map'.

260 |>                 Array.foldLeft(match (term, i) -> match term {
261 |>                     case BodyTerm.Var(var) => match denotation {
262 |>                         case Denotation.Relational =>
263 |>                             Map.insertWith(_ -> t -> t, var, RamTerm.RowLoad(rowVar, i))
264 |>                         case Denotation.Latticenal(_, _, _, glb) =>
265 |>                             if (i < Array.length(terms) - 1)
266 |>                                 Map.insertWith(_ -> t -> t, var, RamTerm.RowLoad(rowVar, i))
267 |>                             else
268 |>                                 let f = t1 -> t2 -> RamTerm.Meet(glb, t1, t2);
269 |>                                 Map.insertWith(f, var, RamTerm.LoadLatVar(rowVar))
270 |>                     }
271 |>                     case _ => x -> x
272 |>                 }, acc)

mismatched types.

Type One: (?result8801807 -> (t8801810 ->{b8801811} ?result8801807)) -> (?result8801807 -> (Array[t8801810] ->{Impure} ?result8801807))
Type Two: (((BodyTerm[t8801848], Int32)) -> (Map[VarSym, RamTerm[t8808787]] -> Map[VarSym, RamTerm[t8808787]])) ->{?eff8801806} ?result8801805


```